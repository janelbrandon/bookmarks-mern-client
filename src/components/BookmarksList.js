import React from 'react'
import Bookmark from './Bookmark'

export default (props) => {
  const {bookmarks} = props
  return (
    <div className="bookmarksList">
      <h1>Bookmarks</h1>
      <ul> {
       bookmarks.map(
        bookmark => <Bookmark key={bookmark._id} {...bookmark}/>
      )}
      </ul>
    </div>
  )
}
