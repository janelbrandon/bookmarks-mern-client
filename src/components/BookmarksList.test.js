import React from 'react'
import BookmarksList from './BookmarksList'
import renderer from 'react-test-renderer'


describe('BookmarksList', () => {
  it('should render as we expect', () => {
    const tree = renderer.create(
    <BookmarksList bookmarks={[{_id:'123',title:'My Bookmark',url:'http://mybookmark.com'}]}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
