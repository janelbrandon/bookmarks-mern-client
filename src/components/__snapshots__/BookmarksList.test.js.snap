// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarksList should render as we expect 1`] = `
<div
  className="bookmarksList"
>
  <h1>
    Bookmarks
  </h1>
  <ul>
     
    <li>
      My Bookmark
       (
      <a
        href="http://mybookmark.com"
        target="_blank"
      >
        Visit
      </a>
      )
      <button
        onClick={[Function]}
      >
        Delete!
      </button>
    </li>
  </ul>
</div>
`;
